README.org

```diff
@@ -47,15 +47,20 @@
 
 ** Spacemacs
    [[https://github.com/emacs-lsp/lsp-mode][lsp-mode]] is included in spacemacs. Add ~lsp~ to ~dotspacemacs-configuration-layers~ and configure the language that you want to use to be backed by ~lsp~ backend.
-
 * Configuration
-  Add the following line in your configuration file where ~programming-mode-hook~ stands for the language specific hook like ~python-mode-hook~.  In addition, ~lsp-mode~ will automatically detect and configure [[https://github.com/emacs-lsp/lsp-ui][lsp-ui]] and [[https://github.com/tigersoldier/company-lsp][company-lsp]]. To turn off that behavior you could set ~lsp-auto-configure~ to ~nil~.
+  Add the following line in your configuration file.
+  #+BEGIN_SRC emacs-lisp
+    (require 'lsp)
+    (add-hook 'prog-mode-hook 'lsp)
```

Comment:
Even a warning `language server ... is not available for some mode` may be annoying for some people.

Reply:
This option will be for initial configuration and targeted at beginners which are looking for zero configuration. Once you find the message irritating you can change it to work as you want(e. g. attach to the proper mode hook). Of course we may restructure the README based on the user feedback.

Reply:
Beginners may not know what `prog-mode-hook` is and this suggestion does not provide any convenience.

Reply:
People choose language servers very conservatively, as the status quo tools are not necessarily worse than LSP counterparts.